### ISSUE TITLE
Halt all feature development until someone is doing at least patch releases

### ISSUE THREAD

**Issue by adriancole** [OpenFeign/feign#646]
Our issues list is backing up, and some of the requested change will have maintenance impact, which there's also little help on. We need to decide whether to continue this project at all. It matters not if folks want it, if no-one is ready to do it.

Personally, I have no time to merge major features, sadly. I won't merge things that are not in (what I consider to be) good shape or create new classes of bugs. This puts me in a bad spot because often change needs significant multi-day effort to get into such a state. I don't have the time or will to do this anymore for reasons including being overcommitted. When I do, it makes me even more stressed out, so sorry I can't do it anymore. It isn't you, it's me.

So, what's the impact? Someone else will be either merging and releasing code, or we should close the project or let it move somewhere else. I cannot offer help merging change anymore, but I can offer help in getting someone access and teaching them how to do that.

For folks who are up to the task! HACKING file roughly describes the culture which effective java also influences. This was put together to prevent the largest amount of bugs and the least api breakage when introducing change. This is why spring was able to include feign in project work like spring cloud (not fear apis will break or change willy nilly). If this project is resumed, and the change culture is otherwise, bear in mind that this may limit the ability for others to support feign. This may nullify some of the goals of even proceeding. For this reason, whoever that takes over (if it is taken over), will need to think carefully about this, as damage caused by not well planned change or interface design have knock-on effects to anyone using the project. The ideal leaders of this project will be very nervous, especially about public facing api change or features that require java versions, or careful resource management. A different change culture can also be the case, but if that is I'd recommend a different group ID so that others aren't broken when things are merged otherwise.

The TL;DR; frankly, this project is at a point of stagnation; I won't be running in gallantly to save it. Someone else needs to be that. If not, we should do as sometimes is done in apache. Set a timeout for attic'ing the project. If that happens attic it and recommend an alternative such as retrofit which is still quite healthy.

Over to you.. If you have opinions on the matter or impact, please reply back.

**Comment by spencergibb** [comment]
Thanks @adriancole. While I have commit and merge rights, I am in a similar situation as Adrian.

**Comment by spencergibb** [comment]
https://twitter.com/springcloud/status/970505041135177733 call for maintainers

**Comment by adriancole** [comment]
PS something that comes up very often and should be noted. Changes that accidentally raise the java level are easy to make. This project has always been JRE 6+, but it isn't 100% about JRE (ex old android code, too). Apis like AutoCloseable are easy to forget were added later. It is often very easy to avoid bumping the entire library to a higher level to achieve a feature, but it isn't always intuitive how or why. The "rule of thumb" has often been don't raise the java level beyond dependencies. For example, if making jackson integration, don't raise the java level higher than what's required for jackson. That's the most portable way regardless of your opinion on whatever version that is.

Folks who again keep this in mind and downstream impact who have assumptions based on language level will be able to proceed without breaking others, often with little impact (ex in the autocloseable case, usually closeable is fine).

This isn't a discussion specifically about one api, just rather the preference which has been to break the least amount of people possible and to consider compatibilities you don't currently have a requirement personally to maintain. This has been in mind with every feature in the past, and changing this will change that culture a bit, perhaps making change easier at the expense of accidentally breaking people who use things in a way you don't expect.

**Comment by waza-kaunda** [comment]
Would definitely like to help. Not blessed with many hours in the day but can help clearing issues. Can't handle the power to commit and merge, too much responsibility to take on currently. Just point me in the right direction on where I can best fit and I'll take what I can to make a difference.

**Comment by alexmarques** [comment]
Hi there, I'm looking to get into open source projects and give back to the community.  I'm willing to help in the project. I'm a java developer with over 10 years of experience in developing web apps. If that sounds good get in touch with me I will be glad to help. Im a full time developer but I can make some time to help in the project.

**Comment by vlotar** [comment]
@adriancole , I'd be glad start contributing as well.
Feign is used heavily in my organization, so I'd like to pay back.
What are the next steps?

**Comment by kdavisk6** [comment]
@adriancole I am also available to help.  We just started using Open Feign and I think it will only grow at my organization as well.

**Comment by foezor** [comment]
Hello i will find some time to help, i used feign for a long time in my work, and i will be glad to payback all the time i saved.

**Comment by TYsewyn** [comment]
Let me know where I can help out @adriancole! I'll be glad to help.

**Comment by adriancole** [comment]
Thanks for the support folks. @velo is helping with release parts. I am
sure he and others could use a hand in clearing the pull request and issue
queue for easy ones.

So concretely, we need folks to get pull requests prioritized and into
shape or closed otherwise.

https://github.com/OpenFeign/feign/issues/643 tracks the next release.

**Comment by adriancole** [comment]
Other concrete things. We will need another person who will do releases
besides @velo.

Ideal case is folks help review other people's issues and pull requests
until they are in mergable shape. I will help @velo or others do the merge
clicking, but it is important the review comes from the community and it is
more than "looks great" ;) even great things often miss additions to
CHANGES or javadoc sometimes tests or formatting boring stuff.


At some point now or later, how the project is managed is on the table. For
example, Netflix years ago were keen to have this break out to where it is
or even Apache. If there is a new community who are interested in taking it
forward, a champion to regroup into apache could be a smart idea. Doing
something like this at a period of stability (vs fast change) is ideal, but
only if there is community support. This could be a separate issue, so if
someone were interested (possibly @johnament) please raise an issue and
rally support for an answer. Carrying on here is fine too obviously.

**Comment by raghuramgururajan** [comment]
@adriancole I am interested in participating in open source contribution opportunity.I am a Java developer with 12+ years of experience

**Comment by adriancole** [comment]
folks who are looking to get involved in OSS via feign... well this is awesome. Highest value is looking at the big backlog of issues and/or stackoverflow and clearing out things marked as questions.

This is a part of triage needed to get things back into shape. Specific tasks beyond that are likely best hashed out on gitter https://gitter.im/OpenFeign/feign

Ex first thing spring cleaning. then look for easy to close issues etc. Those with experience releasing open source feign extensions should help steer what goes into the next release. Anyone can gain experience through our dilemma though :)

**Comment by adriancole** [comment]
heh ps I forgot that I had a similar request of the same nature a few months back https://github.com/OpenFeign/feign/issues/606 I will close that out in favor of this

**Comment by adriancole** [comment]
Note about @velo who is now a contributor:

I added @velo as a contributor, someone notably familiar with feign due to having built several tools on top of it https://github.com/velo?utf8=%E2%9C%93&tab=repositories&q=feign&type=&language=

Marvin lives in "Welly, coolest little city" otherwise known as Wellington, NZ. Keep this timezone in mind :)

His first efforts you may not have noticed are around rehabilitating the build. Accordingly, Marvin will be pulling the trigger on the next release when it is ready #643. What goes into that release is a mixed responsibility as mentioned before.

Many thanks Marvin for stepping up and fingers crossed the momentum continues. Community-driven projects are quite a nice thing to see and be a part of.

**Comment by AjitDas** [comment]
@adriancole we have been using this library and would also like to contribute to the support and maintenance. I will probably start answering some of the easy questions (reported as issues) here and see if we can clean them up. Also I can help with PRs.

**Comment by kdavisk6** [comment]
I'm glad to see more involvement.  Might I make a suggestion, can we nominate someone to help take the lead on assigning out tasks related #643?  I'm afraid that we all my choose something and end up stepping on each other.  Whom shall we talk to to coordinate?  Where should we coordinate? Gitter?  Here?

**Comment by dashaun** [comment]
Hi.  I'm here to help too.

- Got caught up on [gitter](https://gitter.im/OpenFeign/feign)
- Checked out 6 repos ( and failed to build feign-form, feign-annotation-error-decoder, locally)
- Now reviewing the documentation and open issues

**Comment by bahrmichael** [comment]
I'd like to help as well, but feel a bit lost. Labelling the issues with e.g. bug, question, feature, help wanted, etc. would be nice.

I'll join issue discussions for now with what I think makes sense.
